lambda/main.py

```diff
@@ -11,34 +16,36 @@
 logger = logging.getLogger()
 logger.setLevel(logging.INFO)
 
-boto3_session = boto3.Session()
-
 
 def lambda_handler(event, context):
     snapshot_arn = event['detail']['SourceArn']
     event_id = event['detail']['EventID']
 
+    boto3_session = boto3.Session()
     rds_client = boto3_session.client('rds')
-
-    if 'RDS-EVENT-0091' in event_id:
+    
+    instance_id = None
+    snapshot_time = None
+    if event_id in [AUTOMATED_SNAPSHOT_CREATED, MANUAL_SNAPSHOT_CREATED]:
         snapshot = rds_client.describe_db_snapshots(
             DBSnapshotIdentifier=snapshot_arn,
         )['DBSnapshots'][0]
-
         instance_id = snapshot['DBInstanceIdentifier']
         snapshot_time = snapshot['SnapshotCreateTime']
-    else:
+    elif event_id in [AUTOMATED_CLUSTER_SNAPSHOT_CREATED, MANUAL_CLUSTER_SNAPSHOT_CREATED]:
         snapshot = rds_client.describe_db_cluster_snapshots(
             DBClusterSnapshotIdentifier=snapshot_arn,
         )['DBClusterSnapshots'][0]
         instance_id = snapshot['DBClusterIdentifier']
         snapshot_time = snapshot['SnapshotCreateTime']
+    else:
+        raise RuntimeError('Unsupported EventID: {0}'.format(event_id))
```

Comment:
I guess, this means need to do something with `cloudwatch_event_rules` [variable](https://github.com/rallyware/terraform-aws-lambda-rds-s3-export/blob/38d76d4eeab4acbcd3aec8280efb1a3ce8363131/variables.tf#L49). I think need to remove variable `cloudwatch_event_rules` and create new one with the options to enable/disable event rules, something like:
```hcl
{
  manual_snapshot_created_enabled = true 
}
```
What do you think?

Reply:
I agree.